cardano-api/src/Cardano/Api/ProtocolParameters.hs

```diff
@@ -1401,262 +1252,184 @@ unbundleProtocolParams (BundleAsShelleyBasedProtocolParameters _ pp _) = pp
+                      -> Either String (PParams ledgerera)
+toAlonzoCommonPParams
+    protocolParameters@ProtocolParameters {
+      protocolParamCostModels
+    , protocolParamPrices
+    , protocolParamMaxTxExUnits
+    , protocolParamMaxBlockExUnits
+    , protocolParamMaxValueSize
+    , protocolParamCollateralPercent
+    , protocolParamMaxCollateralInputs
+    } = do
+  ppShelleyCommon <- toShelleyCommonPParams protocolParameters
+  costModels <- toAlonzoCostModels protocolParamCostModels
+  prices <-
+    requireParam "protocolParamPrices" toAlonzoPrices protocolParamPrices
+  maxTxExUnits <-
+    requireParam "protocolParamMaxTxExUnits" Right protocolParamMaxTxExUnits
+  maxBlockExUnits <-
+    requireParam "protocolParamMaxBlockExUnits" Right protocolParamMaxBlockExUnits
+  maxValueSize <-
+    requireParam "protocolParamMaxBlockExUnits" Right  protocolParamMaxValueSize
+  collateralPercent <-
+    requireParam "protocolParamCollateralPercent" Right protocolParamCollateralPercent
+  maxCollateralInputs <-
+    requireParam "protocolParamMaxCollateralInputs" Right protocolParamMaxCollateralInputs
+  let ppAlonzoCommon =
+        ppShelleyCommon
+        & ppCostModelsL           .~ costModels
+        & ppPricesL               .~ prices
+        & ppMaxTxExUnitsL         .~ toAlonzoExUnits maxTxExUnits
+        & ppMaxBlockExUnitsL      .~ toAlonzoExUnits maxBlockExUnits
+        & ppMaxValSizeL           .~ maxValueSize
+        & ppCollateralPercentageL .~ collateralPercent
+        & ppMaxCollateralInputsL  .~ maxCollateralInputs
+  pure ppAlonzoCommon
+
+toAlonzoPParams :: Ledger.Crypto crypto
+                => ProtocolParameters
+                -> Either String (PParams (Ledger.AlonzoEra crypto))
+toAlonzoPParams
+    protocolParameters@ProtocolParameters {
+      protocolParamDecentralization
+    , protocolParamUTxOCostPerWord
+    } = do
+  ppAlonzoCommon <- toAlonzoCommonPParams protocolParameters
+  -- QUESTION? This is strange, why do we need to construct Alonzo Tx with Babbage PParams?
+  -- This feels to me like an issue with the api design, as there should never be such an
+  -- incosistency, because PParams affect the validity of the transaction.
+  d <- case protocolParamDecentralization of
+         -- The decentralization parameter is deprecated in Babbage
+         -- so we default to 0 if no decentralization parameter is found
+         -- in the api's 'ProtocolParameter' type. If we don't do this
+         -- we won't be able to construct an Alonzo tx using the Babbage
+         -- era's protocol parameter because our only other option is to
+         -- error.
+         Nothing -> Right minBound
+         Just dParam -> boundRationalEither "D" dParam
+  -- This is the correct implementation that should be the used instead:
+  -- d <- requireParam "protocolParamDecentralization"
+  --                   (boundRationalEither "D")
+  --                   protocolParamDecentralization
```

Comment:
Why is the correct implementation in a comment?

Reply:
Because I would like to get the answer for the above question first. I don't want to break some behavior downstream that potentially relies on the incorrect implementation

Reply:
Here is the ticket about it: https://github.com/input-output-hk/cardano-node/issues/5093